src/main/java/com/iexec/sms/secret/SecretController.java

```diff
@@ -146,38 +144,28 @@ public ResponseEntity<String> addWeb2Secret(@RequestHeader("Authorization") Stri
     }
 
     @PutMapping("/web2")
-    public ResponseEntity<String> updateWeb2Secret(@RequestHeader("Authorization") String authorization,
+    public ResponseEntity<String> updateWeb2Secret(@RequestHeader String authorization,
                                                    @RequestParam String ownerAddress,
                                                    @RequestParam String secretName,
                                                    @RequestBody String newSecretValue) {
+        if (!SecretUtils.isSecretSizeValid(newSecretValue)) {
+            return ResponseEntity.status(HttpStatus.PAYLOAD_TOO_LARGE).build();
+        }
+
         String challenge = authorizationService.getChallengeForSetWeb2Secret(ownerAddress, secretName, newSecretValue);
 
         if (!authorizationService.isSignedByHimself(challenge, authorization, ownerAddress)) {
             log.error("Unauthorized to updateWeb2Secret [expectedChallenge:{}]", challenge);
             return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
         }
 
-        if (web2SecretsService.getSecret(ownerAddress, secretName).isEmpty()) {
+        try {
+            web2SecretsService.updateSecret(ownerAddress, secretName, newSecretValue);
+            return ResponseEntity.noContent().build();
+        } catch (NoSuchElementException e) {
             return ResponseEntity.notFound().build();
         }
-
-        web2SecretsService.updateSecret(ownerAddress, secretName, newSecretValue);
-        return ResponseEntity.noContent().build();
```

Comment:
Not sure why you removed the `if` block. Do you think it's better to control the application flow via exceptions?

Reply:
The `if` block was removed because there was a double read and a double check with the implementation in `Web2SecretsService`.
In Web2SecretsService, the data was retrieved a second time and no check was performed, which leads to a vulnerability issue in SonarCloud.
In order to:
* fix the implementation
* perform a single read
* keep the method signature

I found that using `Optional#orElseThrow` was the best solution.
Part of the challenge lies in the fact that we have a collection of `Secrets` stored in a `Web2Secrets` and that several checks against null values are required.

I think that in this specific case, it is more cumbersome to not use the exceptions when we have methods allowing to stopping as soon as an error is met.